src/pip/_internal/utils/misc.py

```diff
@@ -728,3 +730,28 @@ def prepare_metadata_for_build_editable(
             config_settings=cs,
             _allow_fallback=_allow_fallback,
         )
+
+
+def merge_config_settings(
+    reqs_settings: Mapping[str, Union[str, List[str]]],
+    cli_settings: Mapping[str, Union[str, List[str]]],
+) -> Dict[str, Union[str, List[str]]]:
+    dd: Dict[str, Union[str, List[str]]] = {}
+    for d in (reqs_settings, cli_settings):
```

Comment:
I think this ordering is as good as the other. Shall we document it? Or document that the order is not guaranteed and may change in a future pip version?

Reply:
Good question. Maybe [here](https://pip.pypa.io/en/stable/reference/requirements-file-format/?highlight=config-settings#per-requirement-options)? And also maybe say a brief word about it in the help text of `--config-settings`?